Core/src/Definitions/CMakeLists.txt

```diff
@@ -2,10 +2,20 @@ target_sources(ActsCore PRIVATE Common.cpp Direction.cpp ParticleData.cpp)
 
 include(ActsCodegen)
 
+if(ACTS_USE_SYSTEM_LIBS)
+    set(_codegen_offline_flags "OFFLINE")
```

Comment:
Is there a way to roll the detection into the `acts_code_generation` function, so we don't have to duplicate this in a number of places?